File: practice_python_program/Array/longest_pelindrom_substring_in_string.py
def longest_palindrome(s):
    if not s:  # agar string khali hai
        return ""
    
    result = ""  # final answer store karenge
    
    for i in range(len(s)):
        # Case 1: Odd length palindrome (center i pe)
        l, r = i, i
        while l >= 0 and r < len(s) and s[l] == s[r]:
            # Agar current palindrome result se bada hai
            if (r - l + 1) > len(result):
                result = s[l:r+1]
            l -= 1  # left ko expand karo
            r += 1  # right ko expand karo
        
        # Case 2: Even length palindrome (center i aur i+1 ke beech)
        l, r = i, i + 1
        while l >= 0 and r < len(s) and s[l] == s[r]:
            if (r - l + 1) > len(result):
                result = s[l:r+1]
            l -= 1
            r += 1
    
    return result

# Test karte hain
test_cases = ["mississippi", "avcccvbgf", "abcdc", "a", "abc"]

for s in test_cases:
    print(f"String: {s} -> Longest Palindrome: {longest_palindrome(s)}")

File: practice_java_program/src/array_java/Longest_pelindrom_substring_in_string.java
package array_java;

class Longest_pelindrom_substring_in_string {

    static String longestPalindrome(String s) {
        if (s == null || s.length() == 0) {
            return "";
        }

        String result = "";  // final answer store karega
        int n = s.length();

        for (int i = 0; i < n; i++) {

            // Case 1: Odd length palindrome (center i pe)
            int l = i, r = i;
            while (l >= 0 && r < n && s.charAt(l) == s.charAt(r)) {
                if (r - l + 1 > result.length()) {
                    result = s.substring(l, r + 1);
                }
                l--;
                r++;
            }

            // Case 2: Even length palindrome (center i aur i+1 ke beech)
            l = i;
            r = i + 1;
            while (l >= 0 && r < n && s.charAt(l) == s.charAt(r)) {
                if (r - l + 1 > result.length()) {
                    result = s.substring(l, r + 1);
                }
                l--;
                r++;
            }
        }

        return result;
    }

    // Test cases
    public static void main(String[] args) {
        String[] testCases = {
            "mississippi",
            "avcccvbgf",
            "abcdc",
            "a",
            "abc"
        };

        for (String s : testCases) {
            System.out.println("String: " + s +
                    " -> Longest Palindrome: " + longestPalindrome(s));
        }
    }
}

